subnets/config.go

```diff
@@ -52,11 +67,28 @@ type Config struct {
 }
 
 func (c *Config) Valid() error {
-	if err := c.ConsensusParameters.Verify(); err != nil {
+	if err := c.ConsensusConfig.Verify(); err != nil {
 		return fmt.Errorf("consensus %w", err)
 	}
 	if !c.ValidatorOnly && c.AllowedNodes.Len() > 0 {
 		return errAllowedNodesWhenNotValidatorOnly
 	}
 	return nil
 }
+
+func (c *ConsensusConfig) Verify() error {
+	if c.SnowballParams != nil {
+		if c.SimplexParams != nil {
+			return errTwoConfigs
+		}
+		return c.SnowballParams.Verify()
+	} else if c.SimplexParams != nil {
+		// rudimentary check
+		if !c.SimplexParams.Enabled {
```

Comment:
Doesn't this being non-nil imply that it's enabled since you can only specify one consensus config?